firedrake/preconditioners/pmg.py

```diff
@@ -602,16 +602,18 @@ def get_readonly_view(arr):
 
 
 @cached({}, key=generate_key_evaluate_dual)
-def evaluate_dual(source, target, alpha=tuple()):
+def evaluate_dual(source, target, derivative=None):
```

Comment:
Can this get a numpy-style docstring? At the very least what is written doesn't make it clear what the types are.